test/e2e/elasticsearch_test.go

```diff
@@ -93,7 +94,41 @@ func (suite *ElasticSearchTestSuite) TestSimpleProd() {
 	ProductionSmokeTest("simple-prod")
 }
 
+func (suite *ElasticSearchTestSuite) TestEsIndexPrefixes() {
+	esIndexCleanerEnabled = false
+	esIndexPrefix := "prefix"
+	name := "test-es-index-prefixes"
+
+	exampleJaeger := getJaegerAllInOne(name)
+
+	// Add an index prefix to the CR before creating this Jaeger instance
+	options := exampleJaeger.Spec.Storage.Options.Map()
+	updateOptions := make(map[string]interface{})
+	for key, value := range options {
+		updateOptions[key] = value
+	}
+	updateOptions["es.index-prefix"] = esIndexPrefix
+	exampleJaeger.Spec.Storage.Options = v1.NewOptions(updateOptions)
+
+	err := fw.Client.Create(context.Background(), exampleJaeger, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
+	require.NoError(t, err, "Error deploying Jaeger")
+	defer undeployJaegerInstance(exampleJaeger)
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, name, 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for deployment")
+
+	// Run the smoke test so indices will be created
+	AllInOneSmokeTest(name)
+
+	// Now verify that we have indices with the prefix we want
+	indexWithPrefixExists(esIndexPrefix+"-jaeger-", true)
+
+	// Turn on index clean and make sure we clean up
+	turnOnEsIndexCleaner(name, exampleJaeger)
+	indexWithPrefixExists(esIndexPrefix+"-jaeger-", false)
+}
+
 func (suite *ElasticSearchTestSuite) TestEsIndexCleaner() {
+	esIndexCleanerEnabled = false
```

Comment:
disable index cleaner in index cleaner test?

Reply:
Yes, this has not changed (I've changed the name and refactored some common code, but the test is the same.)  That was added in https://github.com/jaegertracing/jaeger-operator/commit/18d50778d90c3d5984612ba903f53e4f35a2ea44 because of race conditions in the test.

Previously we would get intermittent errors because the index cleaner would run before any traces were created.